linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -731,278 +694,244 @@ private[emitter] object CoreJSLib {
-                const(result, Apply(genIdentBracketSelect(ctor, "apply"), instance :: args :: Nil)),
-                Switch(typeof(result),
-                    List("string", "number", "boolean", "undefined").map(str(_) -> Skip()) :+
-                    str("symbol") -> Return(instance),
-                    Return(If(result === Null(), instance, result)))
-            )
-          })
+
+          // This basically emulates the ECMAScript specification for 'new'.
+          Block(
+              const(instance, Apply(genIdentBracketSelect(ObjectRef, "create"), ctor.prototype :: Nil)),
+              const(result, Apply(genIdentBracketSelect(ctor, "apply"), instance :: args :: Nil)),
+              Switch(typeof(result),
+                  List("string", "number", "boolean", "undefined").map(str(_) -> Skip()) :+
+                  str("symbol") -> Return(instance),
+                  Return(If(result === Null(), instance, result)))
+          )
         }
-      }
+      ),
 
-      // resolveSuperRef
-      locally {
-        val superClass = varRef("superClass")
-        val propName = varRef("propName")
+      defineFunction2("resolveSuperRef") { (superClass, propName) =>
         val getPrototypeOf = varRef("getPrototypeOf")
         val getOwnPropertyDescriptor = varRef("getOwnPropertyDescriptor")
         val superProto = varRef("superProto")
         val desc = varRef("desc")
-        defineFunction("resolveSuperRef", paramList(superClass, propName), {
-          Block(
-              const(getPrototypeOf, genIdentBracketSelect(ObjectRef, "getPrototyeOf")),
-              const(getOwnPropertyDescriptor, genIdentBracketSelect(ObjectRef, "getOwnPropertyDescriptor")),
-              let(superProto, superClass.prototype),
-              While(superProto !== Null(), {
-                Block(
-                    const(desc, Apply(getOwnPropertyDescriptor, superProto :: propName :: Nil)),
-                    If(desc !== Undefined(), Return(desc), Skip()),
-                    superProto := Apply(getPrototypeOf, superProto :: Nil)
-                )
-              })
-          )
-        })
-      }
 
-      // superGet
-      locally {
-        val superClass = varRef("superClass")
-        val self = varRef("self")
-        val propName = varRef("propName")
+        Block(
+            const(getPrototypeOf, genIdentBracketSelect(ObjectRef, "getPrototyeOf")),
+            const(getOwnPropertyDescriptor, genIdentBracketSelect(ObjectRef, "getOwnPropertyDescriptor")),
+            let(superProto, superClass.prototype),
+            While(superProto !== Null(), {
+              Block(
+                  const(desc, Apply(getOwnPropertyDescriptor, superProto :: propName :: Nil)),
+                  If(desc !== Undefined(), Return(desc), Skip()),
+                  superProto := Apply(getPrototypeOf, superProto :: Nil)
+              )
+            })
+        )
```

Comment:
```suggestion
        Block(
          const(getPrototypeOf, genIdentBracketSelect(ObjectRef, "getPrototyeOf")),
          const(getOwnPropertyDescriptor, genIdentBracketSelect(ObjectRef, "getOwnPropertyDescriptor")),
          let(superProto, superClass.prototype),
          While(superProto !== Null(), Block(
            const(desc, Apply(getOwnPropertyDescriptor, superProto :: propName :: Nil)),
            If(desc !== Undefined(), Return(desc), Skip()),
            superProto := Apply(getPrototypeOf, superProto :: Nil)
          ))
        )
```

Reply:
Several other functions below can be reformatted as well using the same scheme. They've all got their indentation affected anyway, so we might as well use the opportunity to benefit from the new style.